astr303a.py

```diff
@@ -0,0 +1,571 @@
+    plt.annotate('$r_{1/2}$',(rhalf+0.01,func(rhalf,*cdf)-.08))
+    plt.annotate('$L_{1/2}$',(rhalf+0.16,func(rhalf,*cdf)))
+    pp.savefig()
+    plt.clf()
+    
+    return cdf,reff,rhalf,totallum
+
+#Calculate recessional velocity for a redshift z
+def velocity(z):
+    c = 2.99792458E5
+    numer = (((z+1)**2)-1)*c
+    denom = (((z+1)**2)+1)
+    v = numer/denom
+    return v
+
+#Format of dataarray rows at this point:
+#[objID(0),ra(1),dec(2),mag(3),magErr(4),ext(5),z(6),zErr(7),magabs(8),magabserr(9),lum(10),distance(11),peculairv(12)]
+def part1415(dataarray,dcoma,comameanz):
+
+    #get recessional velocity
+    comav = velocity(comameanz)
+
+    #Compute recessional velocities of individual galaxies
+    for i in range(0,len(dataarray)):
+        recev = velocity(dataarray[i][6])
+        pecv = comav-recev
+        dataarray[i].append(pecv)
+
+    #create list of all peculiar velocities
+    pecvslist = []
+    for i in range(0,len(dataarray)):
+        pecvslist.append(dataarray[i][12])
+    
+    #Gather data in bins of width 200km/s. Retrive counts from each bin. 
+    histogram = py.hist(pecvslist,bins=np.arange(-3000, 3000, 200))
+    counts = histogram[0]
+    binvalues = histogram[1]
+    binset = []
+
+    #Create list of peculiar velocities at bin centers
+    for i in range(0,len(binvalues)-1):
+        binset.append(binvalues[i]+100)
+
+    #Get initial parameter guesses (mean,stddev). Get gaussian fit.
+    def gauss(x, a, x0, sigma):
+        return a*np.exp(-(x-x0)**2/(2*sigma**2))
+    stddev = np.std(pecvslist)
+    mean = np.mean(pecvslist)
+    coeffs, pcov = optimize.curve_fit(gauss, binset, counts, p0 = [50, mean, stddev])
+    
+    print('Gaussian fit: '+str(coeffs))
+    print('Velocity dispersion: '+str(coeffs[2])+'km/s')
+    
+    #Plot histogram and gaussian curve.
+    plt.title('Figure 7: Coma Cluster Member Galaxy Peculiar Velocities')
+    plt.xlabel('Peculiar Velocity '+r'$v_{pec}\,(km/s)$', fontsize = 12)
+    plt.ylabel('Number of galaxies', fontsize = 12)
+    plt.xlim(-3000,3000)
+    plt.ylim(0,50)
+    plt.hist(pecvslist,bins=np.arange(-3000, 3000, 200),color='lightgrey',label='Member')
+    plt.plot(binset, gauss(binset, *coeffs),color='black',linestyle='-')
+    pp.savefig()
+    plt.clf()
+    return coeffs,dataarray
+
+#Compute M1/2,L1/2, M/L
+def part161718(cdf,coeffs,reff,rhalf,totallum):
+    mhalf = 3*((coeffs[2]/1.023)**2)*(rhalf*10**6)/(4.5*10**-3)
+    print('Virial mass M1/2: '+str(mhalf)+' M(solar)')
+    lhalf = totallum*(cdf[0]*rhalf**cdf[1] + cdf[2])
+    print('L1/2: '+str(lhalf)+' L(solar)')
+    mlratio = mhalf/lhalf
+    print('M/L: '+str(mlratio))
+    return
+
+######MAIN PROGRAM EXECUTION######
+print('-----------------------------------------------------------------')
+print('                    COMA CLUSTER ANALYSIS')
+print('-----------------------------------------------------------------') 
+dataarray = readData(filename)
+print('')
```

Comment:
You don't need the empty string argument here. `print()` will print a blank line.